libs/methodologies/bold/rule-processors/mass-id/src/check-organic-mass-criteria/check-organic-mass-criteria.processor.ts

```diff
@@ -0,0 +1,112 @@
+import type { EvaluateResultOutput } from '@carrot-fndn/shared/rule/standard-data-processor';
+
+import { isNil } from '@carrot-fndn/shared/helpers';
+import { ParentDocumentRuleProcessor } from '@carrot-fndn/shared/methodologies/bold/processors';
+import {
+  type Document,
+  DocumentCategory,
+  DocumentType,
+  MassSubtype,
+  MeasurementUnit,
+} from '@carrot-fndn/shared/methodologies/bold/types';
+import { RuleOutputStatus } from '@carrot-fndn/shared/rule/types';
+
+import { CheckOrganicMassCriteriaProcessorErrors } from './check-organic-mass-criteria.errors';
+
+export class CheckOrganicMassCriteriaProcessor extends ParentDocumentRuleProcessor<Document> {
+  protected readonly processorErrors =
+    new CheckOrganicMassCriteriaProcessorErrors();
+
+  private get RESULT_COMMENT() {
+    return {
+      APPROVED: 'The Organic MassID criteria is met.',
```

Comment:
I feel like we should give a few more details. What do you think?
```suggestion
      APPROVED: 'The Organic MassID criteria is met: Category, Type, Subtype, Value, and Measurement Unit are valid.',
```

Reply:
https://github.com/carrot-foundation/methodology-rules/pull/147/commits/5c44b5acef26885484a4dbdc1ba72f65a8797fbf